datacite-doi-commons/src/test/java/no/unit/nva/datacite/DataciteTransformServiceTest.java

```diff
@@ -0,0 +1,65 @@
+package no.unit.nva.datacite;
+
+import java.net.URI;
+import java.time.Instant;
+import java.util.List;
+import no.unit.nva.publication.doi.dto.Contributor;
+import no.unit.nva.publication.doi.dto.DoiRequest;
+import no.unit.nva.publication.doi.dto.DoiRequestStatus;
+import no.unit.nva.publication.doi.dto.Publication;
+import no.unit.nva.publication.doi.dto.Publication.Builder;
+import no.unit.nva.publication.doi.dto.PublicationDate;
+import no.unit.nva.publication.doi.dto.PublicationStatus;
+import no.unit.nva.publication.doi.dto.PublicationType;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class DataciteTransformServiceTest {
+
+    private static final String EXAMPLE_ID = "https://example.net/nva/publicationIdentifier";
+    private static final String EXAMPLE_DOI_ID = "http://doi.org/11541.2/124530";
+    private static final String EXAMPLE_INSTITUTION_OWNER = "???";
+    private static final String EXAMPLE_TITLE = "The Matrix";
+    private static final String EXAMPLE_CONTRIBUTOR_ID = "https://example.net/contributor/id/4000";
+    private static final String EXAMPLE_CONTRIBUTOR_NAME = "Brinx";
+    private static final String EXAMPLE_CONTRIBUTOR_ARPID = "989114";
+    public static final Publication INVALID_PUBLICATION = null;
+
+    @Test
+    public void dataciteTransformerServiceReturnsXmlStringFromPublicationDto() {
+        DataciteTransformService service = new DataciteTransformService();
+        Publication publication = createBuilderWithAllFieldsSet().build();
+
+        String xml = service.getXml(publication);
+
+        Assertions.assertNotNull(xml);
+    }
+
+
+    @Test
+    public void dataciteTransformerServiceThrowsRuntimeExceptionOnInvalidPublicationDto() {
+        DataciteTransformService service = new DataciteTransformService();
```

Comment:
This test is a bit problematic:

1. The thrown Exception is a very generic Exception and we do not know test exactly where the exception is thrown. While throwing a Runtime exception is not necessarily bad, not testing the message does not ensure that you are actually testing throwing the exception at the point you want to throw it. Any bug could appear that throws a Runtime exception and the test will pass. So we should ALWAYS test for a specific message and (optionally) a more specialized Exception.  

2. The test is misplaced. This test should be in the handler.  This test is actually saying that the handler will throw a Runtime exception or it will report some kind of error but this test is hidden inside the DataciteTransformerService. I could catch the exception in the handler, change the whole functionality of the handler and the test would still pass. 

The DataciteTransformaitonService is an implementation detail and not a feature. In essence, when we are about implementing the handler we should think:  a)what should the **handler** do when the input is invalid? b) what should the **handler** do when the input is invalid? c) what  the **handler** should do when there is an unexpected error? 

By specifying these things we have multiple benefits:
1) The intention is clear throughout the code and it is us leading the code and not the code leading us.
2) The documentation of what the handler should do is in one class, the handler test.
3) Refactoring does not take much work because we are testing for the behavior of the system and not the  behavior of implementation details.

Reply:
Very good input. I have tried to update the test but the complete behaviour of the datacite client is not known to me yet.